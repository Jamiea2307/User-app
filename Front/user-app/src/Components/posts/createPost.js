import { useState } from "react";
import PostButton from "./postButton";
import EditPost from "./editPost";

const CreatePost = () => {
  const [addPost, setAddPost] = useState(false);

  return addPost ? (
    <EditPost displayAddPost={(value) => setAddPost(value)} />
  ) : (
    <PostButton displayAddPost={(value) => setAddPost(value)} />
  );
};

export default CreatePost;
