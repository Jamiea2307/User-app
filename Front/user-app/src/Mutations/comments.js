import { gql } from "@apollo/client";

export const CREATE_COMMENT = gql`
  mutation createComment($parent: String!, $body: String!) {
    createComment(parent: $parent, body: $body)
  }
`;

export const CREATE_REPLY = gql`
  mutation createReply(
    $parentPost: String!
    $parentComment: String!
    $body: String!
  ) {
    createReply(
      parentPost: $parentPost
      parentComment: $parentComment
      body: $body
    )
  }
`;
