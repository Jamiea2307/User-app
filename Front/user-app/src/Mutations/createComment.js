import { gql } from "@apollo/client";

export const CREATE_COMMENT = gql`
  mutation createComment($parent: String!, $body: String!) {
    createComment(parent: $parent, body: $body)
  }
`;
