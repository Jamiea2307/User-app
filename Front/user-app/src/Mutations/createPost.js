import { gql } from "@apollo/client";

export const CREATE_POST = gql`
  mutation createPost($content: String!) {
    createPost(content: $content) {
      content
    }
  }
`;
