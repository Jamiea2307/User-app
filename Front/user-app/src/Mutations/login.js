import { gql } from "@apollo/client";

export const LOGIN_USERS = gql`
  mutation loginUser($email: String!, $password: String!) {
    loginUser(email: $email, password: $password) {
      email
      password
    }
  }
`;
