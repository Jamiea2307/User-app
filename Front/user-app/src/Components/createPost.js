import { useState } from "react";
import styled from "styled-components";
import { CREATE_POST } from "../Mutations/createPost";
import { useMutation } from "@apollo/client";

const CreatePostContainer = styled.div`
  border: 1px solid black;
  border-radius: 0.5rem;
  width: auto;
  height: 10rem;
  background: #fff;
`;

const PostForm = styled.form``;

const CreatePost = () => {
  const [postContent, setPostContent] = useState("");
  const [createPost] = useMutation(CREATE_POST);

  const createNewPost = (e) => {
    console.log(postContent);
  };

  return (
    <CreatePostContainer>
      <PostForm>
        <input onChange={(e) => setPostContent(e.target.value)} />
        <button
          onClick={(e) => {
            e.preventDefault();
            createNewPost(e);
          }}
        >
          Submit
        </button>
      </PostForm>
    </CreatePostContainer>
  );
};

export default CreatePost;
