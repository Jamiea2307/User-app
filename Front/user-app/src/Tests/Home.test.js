import React from "react";
import renderer from "react-test-renderer";
import Home from "../Components/home";
import { MockedProvider } from "@apollo/client/testing";
import { BrowserRouter as Router } from "react-router-dom";
import { CREATE_POST } from "../Mutations/createPost";

describe("Home", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <MockedProvider mocks={[]}>
          <Router>
            <Home />{" "}
          </Router>
        </MockedProvider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
