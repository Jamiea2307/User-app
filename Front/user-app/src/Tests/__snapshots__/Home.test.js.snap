// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
Array [
  <div
    className="sc-bdfBwQ dHTMLg"
  >
    <div
      className="sc-gsTCUz fJoPy"
    >
      <button
        className="sc-eCssSg"
        onClick={[Function]}
      >
        Create Post
      </button>
    </div>
    <div>
      Loading
    </div>
  </div>,
  " ",
]
`;
